### PR TITLE
[cheerio] Allow user to use Buffer in load()

### PR DESCRIPTION
**Pull request by nwtgck**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - [lib/parse.js#L27](https://github.com/cheeriojs/cheerio/blob/e2f8436170e56190beb0dbbd284065624ca4d329/lib/parse.js#L27)
  - [commit: accept buffer argument to cheerio.load](https://github.com/cheeriojs/cheerio/commit/18b82687c44cc05dbb2a25c1879fad2ec02e0c3a)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/cheerio/index.d.ts [hunk 1]
```diff
@@ -266,7 +268,7 @@ interface CheerioElement {
 }
 
 interface CheerioAPI extends CheerioSelector, CheerioStatic {
-  load(html: string, options?: CheerioOptionsInterface): CheerioStatic;
+  load(html: string | Buffer, options?: CheerioOptionsInterface): CheerioStatic;
```
Comment (nwtgck) [inline types/cheerio/index.d.ts]: I wrote like this by using union type `string | Buffer`. However, the original author may prefer to define separately like the following.

```ts
load(html: string, options?: CheerioOptionsInterface): CheerioStatic;
load(html: Buffer, options?: CheerioOptionsInterface): CheerioStatic;
```

I'd like to change the definition when you suggest.

**Review by SimonSchick — APPROVED** [review]

**Review by AzSiAz — APPROVED** [review]

**Comment by timocov** [comment]
It seems that adding `@types/node` breaks some builds where target is browser and which used to test `enzyme` package (and `@types/enzyme` types for that), because it requires `"@types/cheerio": "*"`.

**Comment by timocov** [comment]
Maybe for compatibility here we need to use something like in react-dom? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/480aadcbb3d1225f1a8a768ecad4724ee454c97a/types/react-dom/server/index.d.ts#L1-L7

instead of using `/// <reference types="node" />` put the following lines:

```ts
declare global {
    interface Buffer {}
}
```

@SimonSchick @AzSiAz

**Comment by cscleison** [comment]
@timocov nwtgck this is causing another side effect too https://github.com/microsoft/TypeScript/issues/40388

**Pull request merged**